notes/lecture22.tex

```diff
@@ -0,0 +1,205 @@
+
+$$\min_{Ax=y} \|x\|_0$$
+
+One option for solving the above non-convex optimization problem is to relax the $l_0$-objective to the convex $l_1$-objective as:
+
+$$\min_{Ax=y} \|x\|_1$$
+
+A variation of the above problem is Basis Pursuit Denoising (BPDN),
+
+\begin{align}
+\min \|x\|_1 \text{ s.t. } \|Ax - y \|_2 \leq \epsilon.
+\end{align}
+
+
+Note that under appropriate assumptions on $A$, $x$ and $y$, we observed that $l_1$-minimization still gives the right answer (RIP, restricted nullspace).
+Sparse linear regression has received a lot of attention and there are many related optimization programs like:
+BPDN (allows for noise)
+$$\min_{\|Ax-y\|_2 \leq s} \|x\|_1$$
+
+Constrained Lasso
+$$\min_{\|x\|_1 \leq \lambda }\|Ax-y\|_2^2 $$
+
+Lagrangian Lasso
+$$\min \|Ax-y\|_2^2 +  \lambda \|x\|_1$$
+
+In our setting, we have \\
+
+\centering{nonconvex constraint} 
+$$\downarrow$$
+\centering{cvx relaxation }
+$$\downarrow$$
+\centering{PGD (Projected Gradient Descent)},
+
+or more directly, nonconvex constraint $\rightarrow$ PGD. 
+
+Here, the Restricted Isometry Property allows optimization over nonconvex sets.
+
+
+Now, we discuss the Iterative Hard Thresholding (IHT) which is known as a projected gradient descent (PGD) for sparse vectors.
+
+Iterative Hard Thresholding (IHT) is used with Gradient Descent for the following problem,
+
+\begin{align}
+f(x) &= \frac{1}{2} \| Ax - y \|_2^2 \\
+\nabla f(x) &= A^{\trans}(Ax - y)
+\end{align}
+
+If it is possible to directly optimize over nonconvex sets, then we don't need convex relaxation.
+
+We want to show that the algorithm IHT outputs a solution. The algorithm is defined as:
+
+\begin{align*}
+\text{IHT}(y, A, t, s) \\
+x^1 &\leftarrow 0 \\
+\text{ for } &i=1 ,\cdots, t, \\
+\tilde{x}^{i+1} &\leftarrow x^i - A^{\trans} (Ax - y) \\
+x^{i+1} &\leftarrow P_S( \tilde{x}^{i+1} ) \\
+\text{ return }& \hat{x} \leftarrow x^{t+1}
+\end{align*}
+
+Note that $P_s$ is a projection onto a set of s-sparse vectors. In the rest of the lecture we will see how we should do this projection and how fast the method is.
+
+RIP: matrix $A$ satisfies the $(s,\delta)$-RIP if for all $s$-sparse vectors we have:
+$$(1-\delta) \|x\|_2^2 \leq \|Ax\|_2^2 \leq (1+\delta)\|x\|_2^2$$
+
+Consequences: For all supports $S$ of size $s$, we have:
+$$\|I-X_s^{\trans} X_s\|_2 \leq \delta$$
+
+Setup: \\
+$y = Ax^* + e$ \\
+$x^*$ is $s$-sparse with support $S^*$ \\ 
+$A$ has ($3s,\frac{1}{4})$-RIP \\
+$e$ is arbitrary noise.
+Then the following holds,
+\begin{theorem}
+$$\|x^{i+1}-x^*\|_2 \leq \frac{1}{2} \|x^{i}-x^*\|_2 + \max_{|S| \leq 3s} \|A_s^{\trans} e\|_2$$
+\end{theorem}
+
+\begin{proof}
+Let $S_i = Supp. (x^i)$ and $ S\prime = S^{i+1} \cup S^i \cup S^*$ (so $|S \prime \leq 3s|$). Then:
```

Comment:
It might be unclear for the reader what "Supp." means in an equation. So let's just say "Let S_i be the support of x^i and let [...]"

Also, the \leq 3s should be outside the | |